### PR TITLE
acceptance test for attempt to modify group share permissions when group sharing is off

### PR DESCRIPTION
**Pull request by haribhandari07**
## Description

Includes acceptance tests for existing group shares and when group sharing is turned off and :
- user tries to edit the permissions of existing shares
- user deletes existing group share
## Related Issue

https://github.com/owncloud/core/issues/35916

## How Has This Been Tested?

CI

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Database schema changes (next release will require increase of minor version instead of patch)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Technical debt
- [x] Tests only (no source changes)

## Checklist:

- [ ] Code changes
- [ ] Unit tests added
- [x] Acceptance tests added
- [ ] Documentation ticket raised: <link> 

## Open tasks:

- [ ] Backport (if applicable set "backport-request" label and remove when the backport was done)

### CODE REVIEW

File: tests/acceptance/features/webUIRestrictSharing/restrictSharing.feature [hunk 1]
```diff
@@ -60,3 +60,25 @@ Feature: restrict Sharing
     When the user shares folder "simple-folder" with user "User One" using the webUI
     And the user re-logs in as "user1" using the webUI
     Then folder "simple-folder (2)" should be listed on the webUI
+
+  Scenario: Editing permissions of existing shares when sharing with groups is forbidded
```
Comment (phil-davis) [inline tests/acceptance/features/webUIRestrictSharing/restrictSharing.feature]:
```suggestion
  Scenario: Editing permissions of existing shares when sharing with groups is forbidden
```

File: tests/acceptance/features/lib/FilesPageElement/SharingDialog.php [hunk 2]
```diff
@@ -331,13 +333,21 @@ public function shareWithGroup(
...
 		$permissions,
 		Session $session
 	) {
-		$xpathLocator = \sprintf(
-			$this->permissionsFieldByUserName, $shareReceiverName
-		);
+		if ($userOrGroup == "group") {
+			$permissionFieldByGroup = $this->permissionsFieldByGroupName;
+			$xpathLocator = \sprintf(
+				$permissionFieldByGroup, $shareReceiverName
```
Comment (phil-davis) [inline tests/acceptance/features/lib/FilesPageElement/SharingDialog.php]: IMO we do not need the local variable at all with this code now.
```suggestion
				$this->permissionsFieldByGroupName, $shareReceiverName
```

File: tests/acceptance/features/lib/FilesPageElement/SharingDialog.php [hunk 3]
```diff
@@ -418,13 +429,21 @@ public function setSharingPermissions(
...
```
Comment (phil-davis) [inline tests/acceptance/features/lib/FilesPageElement/SharingDialog.php]: IMO we do not need the local variable at all with this code now.
```suggestion
				$this->permissionsFieldByGroupName, $shareReceiverName
```

**Review by phil-davis — APPROVED** [review]

**Comment by HariBhandari07** [comment]
Backport on https://github.com/owncloud/core/pull/35944

**Comment by phil-davis** [comment]
drone fails because of some problem with JavaScript dependency loading in core master drone steps.

**Comment by phil-davis** [comment]
Closing - this code now exists in the "new" master that came from stable10

**Pull request closed without merging**